ls1intum/Artemis
# `General`: Allow to search by id in exercise and exam import dialog

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
- As requested by @krusche when importing exercises / exams the id is now also considered in the search.

### Description


- Changed to the query so that exercises / exams are included if id is exact match with search term.
- Renaming and simplifying of methods.

### Steps for Testing
- Check that for import of exercises / exams you can search by id **(exact match)**

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
Added extra tests to cover matching by id

<img width="1434" alt="image" src="https://user-images.githubusercontent.com/29718932/181352961-6287a1c7-68cb-458d-86ab-c0d4ee5d33ad.png">
<img width="2302" alt="image" src="https://user-images.githubusercontent.com/29718932/181353046-e71eb402-2365-4026-821d-30ff4812870b.png">
<img width="1801" alt="image" src="https://user-images.githubusercontent.com/29718932/181353207-74eb8761-73e9-4f0e-875f-bea49aa29974.png">
<img width="2998" alt="image" src="https://user-images.githubusercontent.com/29718932/181353455-b5e6f405-532a-4895-bc73-00c66bec9077.png">








### Screenshots

https://user-images.githubusercontent.com/29718932/181133484-278d9fe0-f749-4637-b571-f19095c3c3fd.mp4
